buildSrc/src/main/kotlin/com/icemachined/buildutils/kotlin-library.gradle.kts

```diff
@@ -41,13 +41,13 @@ kotlin {
     val isMingwX64 = hostOs.startsWith("Windows")
 
     val currentTargetName = when {
-        hostOs == "Mac OS X" -> "macosX64"
-        hostOs == "Linux" -> "linuxX64"
-        isMingwX64 -> "mingwX64"
+        hostOs == "Mac OS X" -> setOf("macosX64", "macosArm64")
+        hostOs == "Linux" -> setOf("linuxX64")
+        isMingwX64 -> setOf("mingwX64")
         else -> throw GradleException("Host OS is not supported in Kotlin/Native.")
     }
 
-    val targetNames = nativeTargets.map{ it.name }.filter { !currentTargetName.equals(it) }.toList()
+    val targetNames = nativeTargets.map{ it.name }.filter { !currentTargetName.contains(it) }.toList()
```

Comment:
@icemachined I think this can be simplified later (remove sourceSets), and just name targets one by one.